docs/src/design/stack/main.md

```diff
@@ -0,0 +1,226 @@
+
+$$
+r_i = \alpha_0 + \alpha_1 \cdot t_{0, i} + \alpha_2 \cdot t_{1, i} + \alpha_3 \cdot t_{2, i}
+$$
+
+Then, when row $i$ is added to the table, we'll update the value in the $p_1$ column like so:
+
+$$
+p_1' = p_1 \cdot r_i
+$$
+
+where, $p_1$ denotes the value of column $p_1$ at the current step and $p_1'$ denotes the value of column $p_1$ at the next step.
+
+Analogously, when row $i$ is removed from the table, we'll update the value in column $p_1$ like so:
+
+$$
+p_1' = \frac{p_1}{r_i}
+$$
+
+The initial value of $p_1$ is set to $1$. Thus, if by the time Miden VM finishes executing a program the table is empty (we added and then removed exactly the same set of rows), $p_1$ will also be equal to $1$.
+
+There are a couple of other rules we'll need to enforce:
+* We can delete a row only after the row has been inserted into the table.
+* We can't insert a row with the same address twice into the table (even if the row was inserted and then deleted).
+
+How these are enforced will be described a bit later.
+
+## Right shift
+If an operation adds data to the stack, we say that the operation caused a right shift. For example, `PUSH` and `DUP` operations cause a right shift. Graphically, this looks like so:
+
+![](https://hackmd.io/_uploads/SkhduAZWc.png)
+
+Here, we pushed value $v_{17}$ onto the stack. All other values on the stack are shifted by one slot to the right and the stack depth increases by $1$. There is not enough space at the top of the stack for all $17$ values, thus, $v_1$ needs to be moved to the overflow table.
+
+To do this, we need to rely on another column: $k_0$. This is a system column which keeps track of the current VM cycle. The value in this column is simply incremented by $1$ with every step.
+
+The row we want to add to the overflow table is defined by tuple $(clk, v1, 0)$, and after it is added, the table would look like so:
+
+![](https://hackmd.io/_uploads/BJ-6mqXJc.png)
+
+The reason we use VM clock cycle as row address is that the clock cycle is guaranteed to be unique, and thus, the same row can not be added to the table twice.
+
+Let's push another item onto the stack:
+
+![](https://hackmd.io/_uploads/SkkytRZW9.png)
+
+Again, as we push $v_{18}$ onto the stack, all items on the stack are shifted to the right, and now $v_2$ needs to be moved to the overflow table. The tuple we want to insert into the table now is $(clk+1, v2, clk)$. After the operation, the overflow table will look like so:
+
+![](https://hackmd.io/_uploads/BkGvcqXJc.png)
+
+Notice that $t_2$ for row which contains value $v_2$ points back to the row with address $clk$.
+
+Overall, the during a right shift we do the following:
```

Comment:
nit: there's an extra word here. I think it should just be "Overall, during a right shift..."

Reply:
Fixed!